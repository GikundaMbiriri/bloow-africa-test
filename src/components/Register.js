import React,{useEffect,useState} from 'react'
import { UserAddIcon } from '@heroicons/react/solid'
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { signupUser } from "../redux/actions/userAction";
import { withRouter } from 'react-router-dom';

function Register(props) {
    const [email,setEmail]=useState('');
    const [username,setUsername]=useState('');
    const [password,setPassword]=useState('');
    const [confirmPassword ,setCpassword]=useState('');
    const handleSubmit=()=>{
      

        const userData = {
            username,
          email,
          password,
          confirmPassword
        };
        props.signupUser(userData, props.history);
    }
  return (
    <>
    <div className='w-4/5 md:w-1/3 shadow-2xl mx-auto mb-10 mt-5'>
<UserAddIcon className="h-12 w-12 mx-auto"/>
<h1 className='text-2xl font-bold pb-3 text-center text-dark-blue'>Register Here</h1>
<h1 className='text-center text-lg'>Remember Everything important</h1>
<div className="flex flex-col justify-center ml-8">
<div className="mb-3 xl:w-96">
    <label for="exampleEmail0" className="form-label inline-block mb-2 text-gray-700"
      >Username input</label>
    <input
      type="text"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      value={username}
      onChange={(event)=>setUsername(event.target.value)}
      id="exampleEmail0"
      placeholder="Username input"
    />
  </div>
  <div className="mb-3 xl:w-96">
    <label for="exampleEmail0" className="form-label inline-block mb-2 text-gray-700"
      >Email input</label
    >
    <input
      type="email"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      id="exampleEmail0"
      value={email}
      onChange={(event)=>setEmail(event.target.value)}
      placeholder="Email input"
    />
  </div>
  <div className="mb-3 xl:w-96">
    <label for="examplePassword0" className="form-label inline-block mb-2 text-gray-700"
      >Password input</label
    >
    <input
      type="password"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      value={password}
      onChange={(event)=>setPassword(event.target.value)}
      id="examplePassword0"
      placeholder="Password input"
    />
  </div>
  <div className="mb-3 xl:w-96">
    <label for="examplePassword0" className="form-label inline-block mb-2 text-gray-700"
      >Confirm Password</label
    >
    <input
      type="password"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      value={confirmPassword}
      onChange={(event)=>setCpassword(event.target.value)}
      id="examplePassword0"
      placeholder="Confirm Password"
    />
  </div>
  <div className="flex space-x-2 justify-center mb-3">
  <button type="button" className="inline-block px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"  onClick={handleSubmit}>Register</button>
</div>

</div>
<div className='mb-3'>
<h1 className='text-center'>Already have an account ?</h1>
<Link to="/login">
<h1 className='text-center text-blue-600 text-sm'>Login here</h1>
</Link>
</div>
</div>
    </>
  )
}

Register.propTypes = {
  user: PropTypes.object.isRequired,
  UI: PropTypes.object.isRequired,
};
const mapStateToProps = (state) => ({
  user: state.user,
  UI: state.UI,
});
export default connect(mapStateToProps, { signupUser })(
  withRouter(Register)
);