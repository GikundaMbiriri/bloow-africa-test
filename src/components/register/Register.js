import React,{useEffect,useState} from 'react'
import { UserAddIcon } from '@heroicons/react/solid'
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { signupUser } from "../../redux/actions/userAction";
import { withRouter } from 'react-router-dom';
import {Loading} from '../Loading';
function Register(props) {
const {errors}=props.UI;

    const [email,setEmail]=useState('');
    const [username,setUsername]=useState('');
    const [password,setPassword]=useState('');
    const [confirmPassword ,setCpassword]=useState('');
    const {loading}=props.UI;
  
    const handleSubmit=()=>{
      

        const userData = {
            username,
          email,
          password,
          confirmPassword
        };
        props.signupUser(userData, props.history);
    }
  return (
    <>
    <div className='w-4/5 md:w-1/3 shadow-2xl mx-auto mb-10 mt-5'>
<UserAddIcon className="h-12 w-12 mx-auto"/>
<h1 className='text-2xl font-bold pb-3 text-center text-dark-blue'>Register Here</h1>
<h1 className='text-center text-lg'>Complete this form correctly</h1>
<div className="flex flex-col justify-center ml-8">
<div className="mb-3 xl:w-96">
    <label htmlFor="Username" className="form-label inline-block mb-2 text-gray-700"
      >Username input</label>
    <input
      type="text"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      value={username}
      onChange={(event)=>setUsername(event.target.value)}
      id="Username"
      data-testid='usernameid'
      placeholder="Username input"
    />
              <h2 className='text-center text-md text-red-600' data-testid='eusername'>{errors?errors.username:''}</h2>

  </div>
  <div className="mb-3 xl:w-96">
    <label htmlFor="Email0" className="form-label inline-block mb-2 text-gray-700"
      >Email input</label
    >
    <input
      type="email"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      id="Email0"
      value={email}
      data-testid='emailid'
      onChange={(event)=>setEmail(event.target.value)}
      placeholder="Email input"
    />
              <h2 className='text-center text-md text-red-600' data-testid='eemail'>{errors?errors.email:''}</h2>

  </div>
  <div className="mb-3 xl:w-96">
    <label htmlFor="Password0" className="form-label inline-block mb-2 text-gray-700"
      >Password input</label
    >
    <input
      type="password"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      value={password}
      onChange={(event)=>setPassword(event.target.value)}
      id="Password0"
      data-testid='passwordid'
      placeholder="Password input"
    />
              <h2 className='text-center text-md text-red-600' data-testid='epassword' >{errors?errors.password:''}</h2>

  </div>
  <div className="mb-3 xl:w-96">
    <label htmlFor="cPassword0" className="form-label inline-block mb-2 text-gray-700"
      >Confirm Password</label
    >
    <input
      type="password"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      value={confirmPassword}
      onChange={(event)=>setCpassword(event.target.value)}
      data-testid='cpasswordid'
      id="cPassword0"
      placeholder="Confirm Password"
    />
              <h2 className='text-center text-md text-red-600' data-testid='econfirmPassword'>{errors?errors.confirmPassword:''}</h2>

  </div>
  <div className="flex space-x-2 justify-center mb-3">
  <button type="button" className="inline-block px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"  data-testid='regbutton'  onClick={handleSubmit}>{loading?<Loading/>:<span>Register</span>}</button>
</div>

</div>
<div className='mb-3'>
<h1 className='text-center'>Already have an account ?</h1>
<Link to="/login">
<h1 className='text-center text-blue-600 text-sm'>Login here</h1>
</Link>
</div>
</div>
    </>
  )
}

Register.propTypes = {
  user: PropTypes.object.isRequired,
  UI: PropTypes.object.isRequired,
};
const mapStateToProps = (state) => ({
  user: state.user,
  UI: state.UI,
});
export default connect(mapStateToProps, { signupUser })(
  withRouter(Register)
);