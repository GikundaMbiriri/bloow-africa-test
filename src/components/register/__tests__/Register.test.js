import React from 'react'
import ReactDOM from 'react-dom'
import Register from '../Register';
import {render as rtlRender, screen} from '@testing-library/react'
import renderer from 'react-test-renderer';
import store from '../../../redux/store';
import {Provider} from 'react-redux';
import { HashRouter as Router } from "react-router-dom";

const render=component=>rtlRender(
    <Provider store={store}>
        <Router>
        {component}
        </Router>
    </Provider>
)
it("it renders without crashing",()=>{
    const div = document.createElement('div');
    render(<Register></Register>,div)
})

it('renders correctly', () => {
    const tree = renderer
      .create(
      <Provider store={store}>
          <Router>
      <Register></Register>
      </Router>
      </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('Register',()=>{
    test('input fields should be filled correctly',()=>{
        render(<Register/>)
        const credentials = {username:'peter', email: 'test@gmail.com', password: 'testpass' ,confirmPassword:'testpass'};
        const usernameInput=screen.getByTestId('usernameid');
        usernameInput.value=credentials.username;
        const emailInput=screen.getByTestId('emailid');
        emailInput.value=credentials.email;
        const passwordInput=screen.getByTestId('passwordid');
        passwordInput.value=credentials.password;
        const cpasswordInput=screen.getByTestId('cpasswordid');
        cpasswordInput.value=credentials.confirmPassword;
     

        expect(usernameInput.value).toBe('peter');

        expect(emailInput.value).toBe('test@gmail.com');
        expect(passwordInput.value).toBe('testpass');
        expect(cpasswordInput.value).toBe('testpass');
    
    })
    test('password and confirm password values are equal',()=>{
        render(<Register/>)
        const credentials = {username:'peter', email: 'test@gmail.com', password: 'testpass' ,confirmPassword:'testpass'};
        const usernameInput=screen.getByTestId('usernameid');
        usernameInput.value=credentials.username;
        const emailInput=screen.getByTestId('emailid');
        emailInput.value=credentials.email;
        const passwordInput=screen.getByTestId('passwordid');
        passwordInput.value=credentials.password;
        const cpasswordInput=screen.getByTestId('cpasswordid');
        cpasswordInput.value=credentials.confirmPassword;
     

      
        expect(passwordInput.value).toEqual(cpasswordInput.value)
    
    })
    test('the Register button displays Register text so long as the component is not loading',()=>{
        render(<Register/>)
      
     

        expect(screen.getByTestId('regbutton')).toHaveTextContent('Register')
     
    
    })
    test('the register button does does not display the Register text when the component is loading',()=>{
        let state = store.getState();
        state.UI['loading']=true;
        render(<Register/>)
            expect(screen.getByTestId('regbutton')).not.toHaveTextContent('Register')
        
        
    })
    test('Error messages do not display when there are no errors',()=>{
        let state = store.getState();
        render(<Register/>)
     
        if(!state.UI.errors){
            expect(screen.getByTestId('eusername')).toHaveTextContent('');
            expect(screen.getByTestId('eemail')).toHaveTextContent('');

            expect(screen.getByTestId('epassword')).toHaveTextContent('');
            expect(screen.getByTestId('econfirmPassword')).toHaveTextContent('')

        }
 
    })
    test('displays username error message correctly', () => {
        let state = store.getState();
        state.UI['errors']={username:'Username field cannot be empty'};

        render(<Register/>) 
        expect(screen.getByTestId('eusername')).toHaveTextContent('Username field cannot be empty');

     })
    test('displays password error message correctly', () => {
        let state = store.getState();
        state.UI['errors']={password:'Password field cannot be empty'};

        render(<Register/>) 
        expect(screen.getByTestId('epassword')).toHaveTextContent('Password field cannot be empty');

     })
    test('displays email error message correctly', () => {
        let state = store.getState();
        state.UI['errors']={email:'Email field cannot be empty'};

        render(<Register/>) 
        expect(screen.getByTestId('eemail')).toHaveTextContent('Email field cannot be empty');

     })
    test('displays confirm password error message correctly', () => {
        let state = store.getState();
        state.UI['errors']={confirmPassword:'Confirm Password field cannot be empty'};

        render(<Register/>) 
        expect(screen.getByTestId('econfirmPassword')).toHaveTextContent('Confirm Password field cannot be empty');

     })

})