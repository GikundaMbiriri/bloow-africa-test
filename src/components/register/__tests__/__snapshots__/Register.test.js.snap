// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-4/5 md:w-1/3 shadow-2xl mx-auto mb-10 mt-5"
>
  <svg
    aria-hidden="true"
    className="h-12 w-12 mx-auto"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
    />
  </svg>
  <h1
    className="text-2xl font-bold pb-3 text-center text-dark-blue"
  >
    Register Here
  </h1>
  <h1
    className="text-center text-lg"
  >
    Complete this form correctly
  </h1>
  <div
    className="flex flex-col justify-center ml-8"
  >
    <div
      className="mb-3 xl:w-96"
    >
      <label
        className="form-label inline-block mb-2 text-gray-700"
        htmlFor="Username"
      >
        Username input
      </label>
      <input
        className="
 form-control
 block
 w-full
 px-3
 py-1.5
 text-base
 font-normal
 text-gray-700
 bg-white bg-clip-padding
 border border-solid border-gray-300
 rounded
 transition
 ease-in-out
 m-0
 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
 "
        data-testid="usernameid"
        id="Username"
        onChange={[Function]}
        placeholder="Username input"
        type="text"
        value=""
      />
      <h2
        className="text-center text-md text-red-600"
        data-testid="eusername"
      >
        
      </h2>
    </div>
    <div
      className="mb-3 xl:w-96"
    >
      <label
        className="form-label inline-block mb-2 text-gray-700"
        htmlFor="Email0"
      >
        Email input
      </label>
      <input
        className="
 form-control
 block
 w-full
 px-3
 py-1.5
 text-base
 font-normal
 text-gray-700
 bg-white bg-clip-padding
 border border-solid border-gray-300
 rounded
 transition
 ease-in-out
 m-0
 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
 "
        data-testid="emailid"
        id="Email0"
        onChange={[Function]}
        placeholder="Email input"
        type="email"
        value=""
      />
      <h2
        className="text-center text-md text-red-600"
        data-testid="eemail"
      >
        
      </h2>
    </div>
    <div
      className="mb-3 xl:w-96"
    >
      <label
        className="form-label inline-block mb-2 text-gray-700"
        htmlFor="Password0"
      >
        Password input
      </label>
      <input
        className="
 form-control
 block
 w-full
 px-3
 py-1.5
 text-base
 font-normal
 text-gray-700
 bg-white bg-clip-padding
 border border-solid border-gray-300
 rounded
 transition
 ease-in-out
 m-0
 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
 "
        data-testid="passwordid"
        id="Password0"
        onChange={[Function]}
        placeholder="Password input"
        type="password"
        value=""
      />
      <h2
        className="text-center text-md text-red-600"
        data-testid="epassword"
      >
        
      </h2>
    </div>
    <div
      className="mb-3 xl:w-96"
    >
      <label
        className="form-label inline-block mb-2 text-gray-700"
        htmlFor="cPassword0"
      >
        Confirm Password
      </label>
      <input
        className="
 form-control
 block
 w-full
 px-3
 py-1.5
 text-base
 font-normal
 text-gray-700
 bg-white bg-clip-padding
 border border-solid border-gray-300
 rounded
 transition
 ease-in-out
 m-0
 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
 "
        data-testid="cpasswordid"
        id="cPassword0"
        onChange={[Function]}
        placeholder="Confirm Password"
        type="password"
        value=""
      />
      <h2
        className="text-center text-md text-red-600"
        data-testid="econfirmPassword"
      >
        
      </h2>
    </div>
    <div
      className="flex space-x-2 justify-center mb-3"
    >
      <button
        className="inline-block px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"
        data-testid="regbutton"
        onClick={[Function]}
        type="button"
      >
        <span>
          Register
        </span>
      </button>
    </div>
  </div>
  <div
    className="mb-3"
  >
    <h1
      className="text-center"
    >
      Already have an account ?
    </h1>
    <a
      href="#/login"
      onClick={[Function]}
    >
      <h1
        className="text-center text-blue-600 text-sm"
      >
        Login here
      </h1>
    </a>
  </div>
</div>
`;
